File: pulsecheck/lib/models/blood_pressure_reading.dart
/// A blood pressure reading the user typed in after measuring it with an
/// external cuff. PulseCheck never estimates blood pressure itself — no
/// phone camera or fingerprint sensor can do that reliably.
class BloodPressureReading {
  final int? id;
  final DateTime takenAt;
  final int systolic;
  final int diastolic;
  final int? pulse;
  final String? notes;

  const BloodPressureReading({
    this.id,
    required this.takenAt,
    required this.systolic,
    required this.diastolic,
    this.pulse,
    this.notes,
  });

  Map<String, Object?> toMap() => {
        if (id != null) 'id': id,
        'taken_at': takenAt.toIso8601String(),
        'systolic': systolic,
        'diastolic': diastolic,
        'pulse': pulse,
        'notes': notes,
      };

  factory BloodPressureReading.fromMap(Map<String, Object?> map) => BloodPressureReading(
        id: map['id'] as int?,
        takenAt: DateTime.parse(map['taken_at'] as String),
        systolic: map['systolic'] as int,
        diastolic: map['diastolic'] as int,
        pulse: map['pulse'] as int?,
        notes: map['notes'] as String?,
      );

  /// General reference categorization based on published AHA thresholds.
  /// This is informational only — not a diagnosis, and it deliberately
  /// does not use words like "you have hypertension".
  String get referenceCategory {
    if (systolic >= 180 || diastolic >= 120) return 'Crisis range — seek medical care';
    if (systolic >= 140 || diastolic >= 90) return 'Stage 2 range';
    if (systolic >= 130 || diastolic >= 80) return 'Stage 1 range';
    if (systolic >= 120 && diastolic < 80) return 'Elevated range';
    return 'Normal range';
  }
}

File: vitaltrack/Sources/VitalTrackCore/Models/BloodPressureReading.swift
import Foundation

public enum BloodPressureSource: String, Codable, CaseIterable, Sendable {
    case manual
    case bluetoothCuff
    case appleHealthImport
    case csvImport
}

public enum BloodPressureReferenceRange: String, Codable, Sendable {
    case normal = "Normal range"
    case elevated = "Elevated range"
    case stage1 = "Stage 1 range"
    case stage2 = "Stage 2 range"
    case crisis = "Crisis range — seek medical care"
}

/// A blood pressure reading obtained from a real monitor — VitalTrack never
/// estimates blood pressure itself. `source` records where the numbers came
/// from so History/Reports can show it honestly.
public struct BloodPressureReading: Identifiable, Codable, Equatable, Sendable {
    public let id: UUID
    public let takenAt: Date
    public let systolic: Int
    public let diastolic: Int
    public let pulse: Int?
    public let source: BloodPressureSource
    public let notes: String?

    public init(
        id: UUID = UUID(),
        takenAt: Date,
        systolic: Int,
        diastolic: Int,
        pulse: Int? = nil,
        source: BloodPressureSource = .manual,
        notes: String? = nil
    ) {
        self.id = id
        self.takenAt = takenAt
        self.systolic = systolic
        self.diastolic = diastolic
        self.pulse = pulse
        self.source = source
        self.notes = notes
    }

    /// General reference categorization from published AHA thresholds.
    /// Informational only — never phrased as a diagnosis.
    public var referenceRange: BloodPressureReferenceRange {
        if systolic >= 180 || diastolic >= 120 { return .crisis }
        if systolic >= 140 || diastolic >= 90 { return .stage2 }
        if systolic >= 130 || diastolic >= 80 { return .stage1 }
        if systolic >= 120 && diastolic < 80 { return .elevated }
        return .normal
    }
}
